tools/sys_analyzer/sys_analyzer_linux.py

```diff
@@ -0,0 +1,721 @@
+#!/usr/bin/python3
+"""
+
+Copyright (c) 2019 Intel Corporation All Rights Reserved.
+
+THESE MATERIALS ARE PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL INTEL OR ITS
+CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THESE
+MATERIALS, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+File Name: sys_analyzer_linux.py
+
+Abstract: Script for checking the environment for media-driver/Media SDK
+
+Notes: Run ./sys_analyzer_linux.py
+
+"""
+
+import os, sys, platform
+import os.path
+import pkgconfig # To install: "apt install python3-pip && pip3 install pkgconfig" or "python3 -m pip install pkgconfig"
+
+class diagnostic_colors:
+    ERROR   = '\x1b[31;1m'  # Red/bold
+    SUCCESS = '\x1b[32;1m'  # green/bold
+    RESET   = '\x1b[0m'     # Reset attributes
+    INFO    = '\x1b[34;1m'  # info
+    OUTPUT  = ''            # command's coutput printing
+    STDERR  = '\x1b[36;1m'  # cyan/bold
+    WARNING = '\x1b[33;1m'  # yellow/bold
+
+class loglevelcode:
+    ERROR   = 0
+    SUCCESS = 1
+    INFO    = 2
+    WARNING = 3
+
+def print_info( msg, loglevel ):
+
+    """ printing information """
+
+    if loglevel==loglevelcode.ERROR:
+        color = diagnostic_colors.ERROR
+        msgtype=" [ ERROR ] "
+        print( color + msgtype + diagnostic_colors.RESET + msg )
+    elif loglevel==loglevelcode.SUCCESS:
+        color = diagnostic_colors.SUCCESS
+        msgtype=" [ OK ] "
+        print( color + msgtype + diagnostic_colors.RESET + msg )
+    elif loglevel==loglevelcode.INFO:
+        color = diagnostic_colors.INFO
+        msgtype=" [ INFO ] "
+        print( color + msgtype + diagnostic_colors.RESET + msg )
+    elif loglevel==loglevelcode.WARNING:
+        color = diagnostic_colors.WARNING
+        msgtype=" [ WRN ] "
+        print( color + msgtype + diagnostic_colors.RESET + msg )
+
+    return
+
+def run_cmd(cmd):
+    output=""
+    fin=os.popen(cmd+" 2>&1","r")
+    for line in fin:
+        output+=line
+    fin.close()
+    return output
+
+def parse_pciid(pciid):
+    sts=loglevelcode.ERROR
+    gpustr="unknown"
+
+    id_directory={
+    '0402':'HSW GT1 desktop',
```

Comment:
are there any available distro tools with this?

Reply:
What?